library/systemd/src/lib/yast2/service_widget.rb

```diff
@@ -64,11 +64,21 @@ class ServiceWidget
     include Yast::Logger
     include Yast::UIShortcuts
 
+    attr_reader :default_action
+
     # creates new widget instance for given service
     # @param service [Yast2::SystemService,Yast2::CompoundService] service
     def initialize(service)
       textdomain "base"
       @service = service
+      # When the service is active, by default, propose to reload or restart
+      # it after writting the configuration (bsc#1158946)
+      init_default_action
+    end
+
+    def default_action=(value)
+      @default_action = value
+      service.public_send(default_action) if valid_action?(value)
```

Comment:
I do not like at all to already modify the service action before calling the store method.

Reply:
so why you do not have it in store method?

Reply:
I plan to add it, and modify the defaut_action per current action, it needs to modify also the action_items, will commit the changes ASAP